src/binary.js

```diff
@@ -155,52 +177,78 @@ class Writer {
 
         this.writeAdaptiveNumber(strBuffer.length << 1);
 
-        this.bytes.set(strBuffer, this.pos);
-        this.pos += strBuffer.length;
+        if (!strBuffer.length) {
+            return;
+        }
+
+        if (strBuffer.length > this.chunkSize) {
```

Comment:
Current implementation is inefficient in distribution of string bytes across chunks. Let's say we have a chunkSize = 100 and a string length is 50 bytes, then for pos >= 50 it will close current chunk at pos and create a new chunk to set 50 bytes. However, a better way to set (100-pos) bytes to current chunk, close it and set the rest to a new chunk. For a case when a string length it greater than a chunk size, there are will be more chunks than needed. I'm suggesting the following implementation:

```js
const currentChunkPortion = Math.min(strBuffer.length, this.bytes.length - this.pos);

// set to the current chunk as many bytes as possible
this.bytes.set(strBuffer.subarray(0, strPos), this.pos);

if (currentChunkPortion === strBuffer.length) {
    this.pos += currentChunkPortion;
    return;
}

this.flushChunk();

// create chunks from str buffer with a length equals to chunk size
for ((strBuffer.length - strPos) >= this.chunkSize) {
    this.chunks.push(strBuffer.subarray(strPos, strPos += this.chunkSize))
}

// write the rest
this.createChunk();
if (strPos < strBuffer.length - 1) {
    this.bytes.set(strBuffer, strPos);
}
```